xbmc/cores/AudioEngine/Sinks/pipewire/AESinkPipewire.cpp

```diff
@@ -409,6 +477,36 @@ bool CAESinkPipewire::Initialize(AEAudioFormat& format, std::string& device)
       static_cast<pw_stream_flags>(PW_STREAM_FLAG_AUTOCONNECT | PW_STREAM_FLAG_INACTIVE |
                                    PW_STREAM_FLAG_MAP_BUFFERS | PW_STREAM_FLAG_DRIVER);
 
+  if (!passthrough)
+  {
+    spa_audio_info_raw info{};
+    info.format = pwFormat;
+    info.flags = SPA_AUDIO_FLAG_NONE;
+    info.rate = format.m_sampleRate;
+    info.channels = static_cast<uint32_t>(pwChannels.size());
+
+    for (size_t index = 0; index < pwChannels.size(); index++)
+      info.position[index] = pwChannels[index];
+
+    params.emplace_back(spa_format_audio_raw_build(&builder, SPA_PARAM_EnumFormat, &info));
+  }
+
+  if (passthrough)
+  {
+    spa_audio_info_iec958 info{};
+    info.codec = AEStreamInfoDataTypeToPWIEC958Codec(format.m_streamInfo.m_type);
+    info.flags = SPA_AUDIO_FLAG_NONE;
+    info.rate = (format.m_streamInfo.m_type == CAEStreamInfo::STREAM_TYPE_EAC3)
```

Comment:
Sorry. I was wrong here. Pipewire would postmultiply it, if you gave it the m_streamInfo.m_sampleRate which is 44100 or 48000, but would be entirely fine if you set it to format.m_sampleRate. AE instead always the already post multiplied value to format.m_sampleRate.

Therefore you can make it easy:
Initialize the sink with format.m_sampleRate AND give format.m_sampleRate to Pipewire.

Reply:
I think this is resolved also.